src/libraries/Microsoft.PowerFx.Connectors/Public/CdpExtensions.cs

```diff
@@ -1,22 +1,23 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+using System.Collections.Generic;
+using System.Linq;
 using Microsoft.PowerFx.Types;
 
 namespace Microsoft.PowerFx.Connectors
 {
     public static class CdpExtensions
     {
-        public static bool TryGetFieldExternalTableName(this RecordType recordType, string fieldName, out string tableName, out string foreignKey)
+        public static bool TryGetFieldRelationships(this RecordType recordType, string fieldName, out IEnumerable<ConnectorRelationship> relationships)
```

Comment:
>TryGetFieldRelationships [](http://example.com/codeflow?start=27&length=24)

I'll add [Obsolete]

Reply:
This API should not exist but Tahir's team is currently using it here [https://powerbi.visualstudio.com/Insights%20Apps%20Platform/_git/Platform.Service?path=/src/service/FederatedKnowledge/TabularDatasource/Implementation/DatasetTableProviders/TabularDatasetTableProvider.cs&version=GBmain&_a=contents](https://powerbi.visualstudio.com/Insights%20Apps%20Platform/_git/Platform.Service?path=/src/service/FederatedKnowledge/TabularDatasource/Implementation/DatasetTableProviders/TabularDatasetTableProvider.cs&version=GBmain&_a=contents)
We now know they shouldn't be using it (like TryGetUnderlyingFieldType) and this is due to the recent discovery that we cannot reuse Dataverse "syntax" (a.b.c.d ...) for fields having the relationships in the general CDP case (or "shims")

When Greg will have told us which direction to follow, we'll be able to remove all this code (or replace it with an improved and working version for shim-based tabular connectors)